Nix Fails to Build Against Lowdown 3.0.0

https://github.com/NixOS/nix/blob/b4bd8d39ca3f1394815cf1628995178c045f8ecb/src/libcmd/markdown.cc#L42

[Lowdown 3.0.0 ](https://github.com/kristapsdz/lowdown/releases/tag/VERSION_3_0_0) replaces `LOWDOWN_TERM_NORELLINK` with `LOWDOWN_NORELLINK` causing build failure on Alpine:

```
../src/libcmd/markdown.cc:42:13: error: 'LOWDOWN_TERM_NORELLINK' was not declared in this scope; did you mean 'LOWDOWN_NORELLINK'?
   42 |             LOWDOWN_TERM_NORELLINK // To render full links while skipping relative ones
      |             ^~~~~~~~~~~~~~~~~~~~~~
      |             LOWDOWN_NORELLINK
```

--- a/src/libcmd/markdown.cc
+++ b/src/libcmd/markdown.cc
@@ -38,7 +38,9 @@ static std::string doRenderMarkdownToTerminal(std::string_view markdown)
 #  endif
         .feat = LOWDOWN_COMMONMARK | LOWDOWN_FENCED | LOWDOWN_DEFLIST | LOWDOWN_TABLES,
         .oflags =
-#  if HAVE_LOWDOWN_1_4
+#  if HAVE_LOWDOWN_3
+            LOWDOWN_NORELLINK
+#  elif HAVE_LOWDOWN_1_4
             LOWDOWN_TERM_NORELLINK // To render full links while skipping relative ones
 #  else
             LOWDOWN_TERM_NOLINK

--- a/src/libcmd/meson.build
+++ b/src/libcmd/meson.build
@@ -44,6 +44,10 @@ configdata.set(
   'HAVE_LOWDOWN_1_4',
   lowdown.version().version_compare('>= 1.4.0').to_int(),
 )
+configdata.set(
+  'HAVE_LOWDOWN_3',
+  lowdown.version().version_compare('>= 3.0.0').to_int(),
+)
 
 readline_flavor = get_option('readline-flavor')
 if readline_flavor == 'editline'

